op-node/rollup/derive/attributes.go

```diff
@@ -94,23 +96,37 @@ func (ba *FetchingAttributesBuilder) PreparePayloadAttributes(ctx context.Contex
 	}
 
 	// Sanity check the L1 origin was correctly selected to maintain the time invariant between L1 and L2
-	nextL2Time := l2Parent.Time + ba.cfg.BlockTime
+	nextL2Time := l2Parent.Time + ba.rollupCfg.BlockTime
 	if nextL2Time < l1Info.Time() {
 		return nil, NewResetError(fmt.Errorf("cannot build L2 block on top %s for time %d before L1 origin %s at time %d",
 			l2Parent, nextL2Time, eth.ToBlockID(l1Info), l1Info.Time()))
 	}
 
-	l1InfoTx, err := L1InfoDepositBytes(seqNumber, l1Info, sysConfig, ba.cfg.IsRegolith(nextL2Time))
+	l1InfoTx, err := L1InfoDepositBytes(ba.rollupCfg, sysConfig, seqNumber, l1Info, nextL2Time)
 	if err != nil {
 		return nil, NewCriticalError(fmt.Errorf("failed to create l1InfoTx: %w", err))
 	}
 
+	// If this is the Ecotone activation block we update the system config by copying over "Scalar"
+	// to "BasefeeScalar". Note that after doing so, the L2 view of the system config differs from
+	// that on the L1 up until we receive a "type 4" log event that explicitly updates the new
+	// scalars.
+	if ba.rollupCfg.IsEcotoneActivationBlock(nextL2Time) {
```

Comment:
specs don't quite capture this yet, but I see that's fixed in https://github.com/ethereum-optimism/optimism/pull/8832/files